### DISCUSSION TITLE
Extracting Arabic embeddings

### DISCUSSION THREAD

**Discussion by juman98** [aub-mind/arabert#94] [General]
Hello, I'm new to transformers, I'm trying to extract the embeddings for a custom dataset, I'm not sure if the implementation is right and honestly I'm not sure if this is normal to happen, I was trying to retrieve the last hidden state so I could use them later on, so could the problem be the implementation of my whole code, or could it actually be as it said?

Here's the snippet of the code: 
![image](https://user-images.githubusercontent.com/54850120/115142796-56d9b900-a04c-11eb-85e6-6f077be0b988.png)

and here's the error:
![image](https://user-images.githubusercontent.com/54850120/115142727-f9456c80-a04b-11eb-94b3-3161ecbe7097.png)

**Comment by WissamAntoun** [comment]
you are probably passing all the data at the same time. I suggest sending the sentences in batches or 1-by-1

**Reply by juman98** [reply]
I actually am passing them all, could you suggest me away for using batches?

**Reply by WissamAntoun** [reply]
```python
import more_itertools
from tqdm import tqdm_notebook as tqdm

last_hidden_states = []
for batch in tqdm(more_itertools.chunked(sentences, 32)):
  #process the sentences here then pass through the model
  processed_batch = tokenizer(batch, return_tensors 'pt)
  with torch.no_grad():
    last_hidden_states.extend(model(**processed_batch))
```

this should work
